### PR TITLE
add command interface

### PR DESCRIPTION
**Pull request by blu-nitro**
Add command interface
Add full hardware tests (and slow tests) using command

Merge command should work as well

### CODE REVIEW

**Comment by blu-nitro** [comment]
@bot help test
comment

**Comment by blufony** [comment]
@nitrokey-ci full-test

File: .gitlab-ci.yml [hunk 1]
```diff
@@ -209,13 +217,70 @@ hardware-tests:
...
+  after_script: []
+  script:
+    - >
+      if [ "$FULL_TEST_STATUS" == "failed" ]; then
+        echo Full hardware tests completed with errors: > report.txt
+      else
+        echo Full hardware tests completed successfully: > report.txt
+      fi
+    - echo https://${CI_SERVER_HOST}/${CI_PROJECT_NAMESPACE}/${CI_PROJECT_NAME}/-/pipelines/${CI_PIPELINE_ID}/test_report >> report.txt
+    - nitrokey-ci write-comment --owner Nitrokey --repo nitrokey-3-firmware --id ${CI_JOB_NAME}_${COMMENT_ID} --commit $(git rev-parse HEAD) report.txt
```
Comment (robin-nitrokey) [inline .gitlab-ci.yml:285]: Where is `$[COMMENT_ID}` set?

**Review by robin-nitrokey — APPROVED** [review]
Looks good!

I think `report` would be a more intuitive stage name, especially as it is also used for the job name, but `com-interface` is fine too.

**Comment by blu-nitro** [comment]
> I think `report` would be a more intuitive stage name, especially as it is also used for the job name, but `com-interface` is fine too.

There are already other commands in the making and for some `report` would not be fitting. Of course this is independent of the command interface itself, so we could easily add a stage called `report` for such jobs, i just didn't think it was worth the clutter and if you say it's fine like this i would leave it for now. If at some point we find ourselves wanting to separate reports from commands, it's very easy to change.

**Comment by robin-nitrokey** [comment]
Then maybe just `commands`?  My problem with `com-interface` is that `com` is not a very common abbreviation (at least for me) and `interface` is so generic that it is hard to understand the term unless you already know its meaning.

File: .gitlab-ci.yml (hunk 1 above)
Comment (blu-nitro) [reply .gitlab-ci.yml:285]: This is set in [nitrokey-ci](https://github.com/Nitrokey/nitrokey-ci/blob/934fe3d441cfd15901a88aa5aff712edd29a78aa/src/api.rs#L160C5-L160C48) when any command pipeline is triggered.

**Comment by blu-nitro** [comment]
> Then maybe just `commands`? My problem with `com-interface` is that `com` is not a very common abbreviation (at least for me) and `interface` is so generic that it is hard to understand the term unless you already know its meaning.

Good point, i suppose `com-interface` was the internal name for it, but using something more expressively would be great for public use. I changed it now here and in `github_commands.yml`.

**Comment by robin-nitrokey** [comment]
@nitrokey-ci merge

**Comment by robin-nitrokey** [comment]
@nitrokey-ci full-test

**Comment by sosthene-nitrokey** [comment]
@nitrokey-ci full-test

**Pull request merged**